src/ids-host.test.ts

```diff
@@ -0,0 +1,326 @@
+import {
+  IdsHost,
+  discoverServicesStub,
+  mungeDiagnosticEndpoint,
+  orderRecordsByPriorityWeight,
+  recordToUrl,
+  weightedRandom,
+} from "./ids-host";
+import { SrvRecord } from "node:dns";
+import { assert, describe, expect, test } from "vitest";
+
+function mkRecord(
+  weight: number,
+  priority = 0,
+  suffix = "install.determinate.systems",
+): SrvRecord {
+  return {
+    weight,
+    priority,
+    port: priority * 100 + weight,
+    name: `${priority}.${weight}.${suffix}`,
+  };
+}
+
+async function mkPromise<T>(lookup: () => T): Promise<T> {
```

Comment:
I'm not sure why this is necessary. It seems to be for testing purposes? But Vitest does enable you to test async functions.